cmd/hln/cmd/get_ingress.go

```diff
@@ -0,0 +1,121 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	watchtools "k8s.io/client-go/tools/watch"
+
+	"github.com/h8r-dev/heighliner/internal/k8sfactory"
+)
+
+const (
+	defaultNS  = "ingress-nginx"
+	defaultSVC = "ingress-nginx-controller"
+	igLabel    = "app.kubernetes.io/component=controller"
+	defaultIP  = "127.0.0.1" // This IP is for local kind and minikube
+)
+
+type getIngressOptions struct {
+	genericclioptions.IOStreams
+}
+
+func newGetIngressCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := &getIngressOptions{
+		IOStreams: streams,
+	}
+	cmd := &cobra.Command{
+		Use:   "ingress",
+		Short: "Get ingress IP address",
+		Args:  cobra.NoArgs,
+		RunE:  o.runGetIngress,
+	}
+
+	return cmd
+}
+
+func (o *getIngressOptions) runGetIngress(cmd *cobra.Command, args []string) error {
+	defaultTimeout := 90 * time.Second
+	if err := waitForIGController(defaultNS, igLabel, defaultTimeout); err != nil {
+		return err
+	}
+	ip, err := getIngressIP(defaultNS, defaultSVC)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(o.Out, ip)
+	return nil
+}
+
+func waitForIGController(namespace, label string, timeout time.Duration) error {
+	endTime := time.Now().Add(timeout)
+	cs, err := k8sfactory.GetDefaultClientSet()
+	if err != nil {
+		return err
+	}
+	for {
+		timeout = time.Until(endTime)
```

Comment:
Maybe print something during each wait? Otherwise it will block out for 90s and confuse users